modin/pandas/base.py

```diff
@@ -3127,6 +3127,42 @@ def __init__(
         ]
         self.__groups = self.__get_groups(*self.resample_args)
 
+    def __getitem__(self, key):
+        """
+        Get ``Resampler`` based on `key` columns of original dataframe.
+
+        Parameters
+        ----------
+        key : str or list
+            String or list of selections.
+
+        Returns
+        -------
+        modin.pandas.BasePandasDataset
+            New ``Resampler`` based on `key` columns subset
+            of the original dataframe.
+        """
+
+        def _get_new_resampler(key):
+            subset = self._dataframe[key]
+            resampler = type(self)(subset, *self.resample_args)
+            return resampler
```

Comment:
```suggestion
            return Resampler(self._dataframe[key], *self.resample_args)
```
I don't see a point in doing 3 lines instead of one here, unless you expect this method to be inherited by some child class?.. I mean this `type(self)(args)` thing looks cool but somewhat hacky.

Reply:
agree

Reply:
We might want to have a child class (like we have Dataframe(BasePandasDataset)) that inherits the method so I think it is fine.